pkg/endpoint/owner.go

```diff
@@ -89,6 +89,15 @@ type Owner interface {
 	// isCTLocal should be set as true if the endpoint's CT table is either
 	// local or not (if is not local then is assumed to be global).
 	FlushCTEntries(e *Endpoint, isCTLocal bool, ips []net.IP, idsToKeep policy.RuleContexts)
+
+	// UpdateSecLabels add and deletes the given labels on given endpoint ID.
+	// The received `add` and `del` labels will be filtered with the valid label
+	// prefixes.
+	// The `add` labels take precedence over `del` labels, this means if the same
+	// label is set on both `add` and `del`, that specific label will exist in the
+	// endpoint's labels.
+	// On success, returns 0, nil. Otherwise, returns an API error code and error msg.
+	UpdateSecLabels(id string, add, del labels.Labels) (int, error)
```

Comment:
I'm on the fence about this, mostly because UpdateSecLabels (and the various Daemon methods it calls) doesn't seem relevant to the owner of an endpoint. In particular, it seems like this function should be on the endpoint itself but that's a non-trivial refactor from what I can tell :/

Reply:
I agree, but also this PR was getting big enough as it is that I didn't really want to try to tackle this at this stage.

Reply:
@aanm Mostly just because Identity management is all in the daemon currently, and the caller isn't in the daemon package so it needs to deal with the abstract class "Owner" rather than knowing the internal details of the daemon. If we refactoring all of the daemon `DeleteIdentity()`, `updateEndpointIdentity()`, etc. stuff out of daemon, then it wouldn't need to be here. We can do that in a follow up PR.